src/esp/bindings/bindings.cpp

```diff
@@ -359,38 +360,18 @@ PYBIND11_MODULE(habitat_sim_bindings, m) {
                           py::object traceback) { self->renderExit(); })
       .def(
           "read_frame_rgba",
-          [](RenderingTarget& self,
-             Eigen::Ref<Eigen::Matrix<uint8_t, Eigen::Dynamic, Eigen::Dynamic,
-                                      Eigen::RowMajor>>& img) {
-            self.readFrameRgba(img.data());
+          [](RenderingTarget& self, const Magnum::PyImageView<2, char>& view) {
+            self.readFrameRgba(view);
           },
-          py::arg("img").noconvert(),
-          R"(
-      Reads RGBA frame into passed img in uint8 byte format.
-
-      Parameters
-      ----------
-      img: numpy.ndarray[uint8[m, n], flags.writeable, flags.c_contiguous]
-           Numpy array array to populate with frame bytes.
-           Memory is NOT allocated to this array.
-           Assume that ``m = height`` and ``n = width * 4``.
-      )")
-      .def(
-          "read_frame_depth",
-          [](RenderingTarget& self,
-             Eigen::Ref<Eigen::Matrix<float, Eigen::Dynamic, Eigen::Dynamic,
-                                      Eigen::RowMajor>>& img) {
-            self.readFrameDepth(img.data());
-          },
-          py::arg("img").noconvert(), R"()")
-      .def(
-          "read_frame_object_id",
-          [](RenderingTarget& self,
-             Eigen::Ref<Eigen::Matrix<uint32_t, Eigen::Dynamic, Eigen::Dynamic,
-                                      Eigen::RowMajor>>& img) {
-            self.readFrameObjectId(img.data());
-          },
-          py::arg("img").noconvert(), R"()")
+          "Reads RGBA frame into passed img in uint8 byte format.")
+      .def("read_frame_depth",
+           [](RenderingTarget& self, const Magnum::PyImageView<2, char>& view) {
+             self.readFrameDepth(view);
+           })
+      .def("read_frame_object_id",
+           [](RenderingTarget& self, const Magnum::PyImageView<2, char>& view) {
```

Comment:
Why is this a `char` here?

Reply:
The basic `ImageView<dimensions, T>` class uses `char`/`const char` to distinguish between mutable and immutable version. The C++ API has nice typedefs for all (`ImageView2D`, `MutableImageView2D` and so on) that but I didn't bother doing that for this python wrapper class because it's used only very rarely. (The `PyImageView` is needed in order to ensure the memory owner is correctly refcounted, btw.)

Reply:
Nope, it doesn't seem like that's needed.  Just wanted to make sure `char` is correct given that the underlying datatype is `uint32_t` :)

Reply:
The image itself is typeless (would be a template hell otherwise). For typed access there's the `pixels()` function ([C++ API](https://doc.magnum.graphics/magnum/classMagnum_1_1ImageView.html#a13bedd03f65287197a040b32ac51ac7f), [Python API](https://doc.magnum.graphics/python/magnum.MutableImageView2D.html#pixels)), returning a n-dimensional array view that you can then cast to a concrete type based on the actual pixel format. Also check the [related blog post](https://blog.magnum.graphics/backstage/multidimensional-strided-array-views/) :)